### ISSUE TITLE
Cannot un-ignore specific subdirectory in node_modules

### ISSUE THREAD

**Issue by tomkur** [eslint/eslint#5410]
I am developing React Native application and I am keeping the code inside ```node_modules/mylibrary```. The reason why I keep it there is to avoid relative paths between modules in my library and that's the recommended way to do that according to [this](https://github.com/substack/browserify-handbook#node_modules). I would use symlinks if it wasn't for [this](https://github.com/facebook/react-native/issues/637).

I would therefore like to lint code in ```node_modules/mylibrary```. Because ```node_modules``` are by default ignored, I've tried adding the following line to ```.eslintignore```, but it didn't help:
```
!node_modules/mylibrary/**/*.*
```

**What version of ESLint are you using?**
v2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
My .eslintrc file looks like this:
```
{
  "extends": ["standard"],
  "parser": "babel-eslint"
}
```
**What did you do? Please include the actual source code causing the issue.**
I've created ```node_modules/mylibrary/abc.js``` with invalid content:
```
abc
```
Basically I am using atom editor and noticed that code is not linted. I confirmed it by calling three commands:
```
./node_modules/.bin/eslint --ignore-path .eslintignore ./node_modules/mylibrary
./node_modules/.bin/eslint ./node_modules/mylibrary
./node_modules/.bin/eslint --no-ignore ./node_modules/mylibrary
```
**What did you expect to happen?**
I would expect the output to be the same for all three commands

**What actually happened? Please include the actual, raw output from ESLint.**
First two commands output nothing (no errors). Third one output:
```
/Users/tomkur/git/myproject/node_modules/mylibrary/abc.js
  1:1  error  'abc' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by ilyavolodin** [comment]
Hmm.. I don't think you can un-ignore anything inside node_modules, because it's ignored at a different level. Not really sure what we can do for this case. @eslint/eslint-team ideas?

**Comment by nzakas** [comment]
Seems like this should work if we are mimicking .gitignore behavior.

**Comment by alberto** [comment]
Current .gitignore behaviour doesn't allow unignoring a subdirectory if the directory is ignored, so it's working as expected according to our ignore patterns.

If we wanted to allow people to unignore, we should change the ignore pattern to `/node_modules/*` (also `/bower_components/*` ?) and from what I saw, we should get rid of `this.ig.default`

**Comment by nzakas** [comment]
@alberto that seems like a good approach

**Comment by IanVS** [comment]
We may want to keep ig.default.  It is used in cli-engine to determine
whether warnings should be shown. But, maybe we can handle that another
way.

**Comment by IanVS** [comment]
I'm doing some work in the ignore rules already, happy to take this one on if you haven't started, @alberto.

**Comment by alberto** [comment]
Ok, go for it.

**Comment by alberto** [comment]
Notice there is a bug in `ignore`: https://github.com/kaelzhang/node-ignore/issues/10

I don't think we should rely on that behaviour.

**Comment by kaelzhang** [comment]
fixed in kaelzhang/node-ignore#11, I will create a pr soon. Now node-ignore acts exactly as the gitignore spec does, according to every rule.

but we should use a default ignore pattern with `node_modules/*` instead of `node_modules`

**Comment by alberto** [comment]
Great! Thanks @kaelzhang

**Comment by IanVS** [comment]
This is a little trickier than I originally hoped, due to the way our default and custom ignore rules are separated.  After https://github.com/eslint/eslint/pull/5461 is merged I will tackle this issue as a separate PR.

**Comment by tomkur** [comment]
Awesome, thanks!
